diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 7b19efc9b8..2f6e66c1cd 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -101,7 +101,8 @@ dtb-$(CONFIG_ARCH_ZYNQ) += zynq-zc702.dtb \
 	zynq-zc770-xm010.dtb \
 	zynq-zc770-xm011.dtb \
 	zynq-zc770-xm012.dtb \
-	zynq-zc770-xm013.dtb
+	zynq-zc770-xm013.dtb \
+	zynq-tjd-sdr.dtb
 dtb-$(CONFIG_ARCH_ZYNQMP) += \
 	zynqmp-ep108.dtb			\
 	zynqmp-zcu100.dtb			\
diff --git a/arch/arm/dts/zynq-tjd-sdr.dts b/arch/arm/dts/zynq-tjd-sdr.dts
new file mode 100644
index 0000000000..09f715409c
--- /dev/null
+++ b/arch/arm/dts/zynq-tjd-sdr.dts
@@ -0,0 +1,84 @@
+/*
+ * Analog Devices Inc. Pluto SDR board DTS
+ *
+ * Copyright (C) 2015-2018 Analog Devices Inc.
+ *
+ * SPDX-License-Identifier:	GPL-2.0+
+ */
+/dts-v1/;
+#include "zynq-7000.dtsi"
+
+/ {
+	model = "Zynq Tjd SDR Board";
+	compatible = "xlnx,zynq-7000";
+
+	aliases {
+		ethernet0 = &gem0;
+		serial0 = &uart1;
+		spi0 = &qspi;
+		mmc0 = &sdhci1;
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0x0 0x20000000>;
+	};
+
+	chosen {
+		bootargs = "earlyprintk";
+		linux,stdout-path = &uart1;
+		stdout-path = &uart1;
+	};
+
+	usb_phy0: phy0 {
+		compatible = "usb-nop-xceiv";
+		#phy-cells = <0>;
+	};
+};
+
+&qspi {
+	status = "okay";
+	is-dual = <0>;
+	num-cs = <1>;
+	flash@0 {
+		compatible = "n25q256a11,";
+		reg = <0x0>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <4>;
+		spi-max-frequency = <50000000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		partition@qspi-fsbl-uboot {
+			label = "qspi-fsbl-uboot";
+			reg = <0x0 0x100000>; /* 1M */
+		};
+		partition@qspi-uboot-env {
+			label = "qspi-uboot-env";
+			reg = <0x100000 0x20000>; /* 128k */
+		};
+		partition@qspi-nvmfs {
+			label = "qspi-nvmfs";
+			reg = <0x120000 0xE0000>; /* 1M */
+		};
+		partition@qspi-linux {
+			label = "qspi-linux";
+			reg = <0x200000 0xE00000>; /* 14M */
+		};
+	};
+};
+
+&uart1 {
+	status = "okay";
+};
+
+&usb0 {
+	status = "okay";
+	dr_mode = "host";
+	usb-phy = <&usb_phy0>;
+};
+
+&sdhci1 {
+	u-boot,dm-pre-reloc;
+	status = "okay";
+};
+
diff --git a/board/xilinx/zynq/board.c b/board/xilinx/zynq/board.c
index 2b23baf83c..2b936c70a8 100644
--- a/board/xilinx/zynq/board.c
+++ b/board/xilinx/zynq/board.c
@@ -113,7 +113,7 @@ int board_late_init(void)
 #include <environment.h>
 int misc_init_r(void)
 {
-#define BUTTON_GPIO 14
+#define BUTTON_GPIO 41
 
 	gpio_request(BUTTON_GPIO, "SWITCH");
 	gpio_direction_input(BUTTON_GPIO);
diff --git a/configs/zynq_tjd_defconfig b/configs/zynq_tjd_defconfig
new file mode 100644
index 0000000000..cd3a580ecf
--- /dev/null
+++ b/configs/zynq_tjd_defconfig
@@ -0,0 +1,64 @@
+CONFIG_ARM=y
+CONFIG_SYS_CONFIG_NAME="zynq_zc70x"
+CONFIG_ARCH_ZYNQ=y
+CONFIG_SYS_MALLOC_F_LEN=0x800
+CONFIG_DEFAULT_DEVICE_TREE="zynq-tjd-sdr"
+CONFIG_SPL=y
+CONFIG_FIT=y
+CONFIG_FIT_VERBOSE=y
+CONFIG_FIT_SIGNATURE=y
+CONFIG_BOOTDELAY=3
+CONFIG_SYS_NO_FLASH=y
+CONFIG_HUSH_PARSER=y
+CONFIG_SYS_PROMPT=Tjd> "
+# CONFIG_CMD_IMLS is not set
+# CONFIG_CMD_XIMG is not set
+# CONFIG_CMD_FLASH is not set
+CONFIG_CMD_SF=y
+CONFIG_CMD_SPI=y
+CONFIG_CMD_USB=y
+CONFIG_CMD_DFU=y
+CONFIG_CMD_GPIO=y
+# CONFIG_CMD_SETEXPR is not set
+# CONFIG_CMD_NET is not set
+# CONFIG_CMD_NFS is not set
+CONFIG_CMD_CACHE=y
+CONFIG_CMD_EXT2=y
+CONFIG_CMD_EXT4=y
+CONFIG_CMD_EXT4_WRITE=y
+CONFIG_CMD_FAT=y
+CONFIG_CMD_FS_GENERIC=y
+CONFIG_OF_EMBED=y
+CONFIG_SPL_DM_SEQ_ALIAS=y
+CONFIG_LED=y
+CONFIG_LED_GPIO=y
+CONFIG_SPI_FLASH=y
+CONFIG_SPI_FLASH_BAR=y
+CONFIG_SPI_FLASH_ISSI=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SPI_FLASH_SPANSION=y
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_ZYNQ_QSPI=y
+CONFIG_USB=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_ULPI_VIEWPORT=y
+CONFIG_USB_ULPI=y
+CONFIG_USB_STORAGE=y
+CONFIG_USB_GADGET=y
+CONFIG_CI_UDC=y
+CONFIG_USB_GADGET_VBUS_DRAW=500
+CONFIG_USB_GADGET_DOWNLOAD=y
+CONFIG_G_DNL_MANUFACTURER="Analog Devices Inc."
+CONFIG_G_DNL_VENDOR_NUM=0x0456
+CONFIG_G_DNL_PRODUCT_NUM=0xb674
+CONFIG_REGEX=y
+CONFIG_LIB_RAND=y
+
+CONFIG_ZYNQ_SDHCI=y
+CONFIG_CMD_MMC=y
+CONFIG_MMC_SDHCI=y
+CONFIG_MMC_SDHCI_ZYNQ=y
+CONFIG_SD_BOOT=y
+DEBUG=y
+
diff --git a/include/configs/zynq-common.h b/include/configs/zynq-common.h
index fd50caf450..3883187c64 100644
--- a/include/configs/zynq-common.h
+++ b/include/configs/zynq-common.h
@@ -60,7 +60,7 @@
 
 /* QSPI */
 #ifdef CONFIG_ZYNQ_QSPI
-# define CONFIG_SF_DEFAULT_SPEED	50000000
+# define CONFIG_SF_DEFAULT_SPEED	10000000
 # define CONFIG_SF_DUAL_FLASH
 #endif
 
@@ -78,12 +78,12 @@
 "dfu_sf_info="\
 "set dfu_alt_info " \
 "boot.dfu raw 0x0 0x100000\\\\;" \
-"firmware.dfu raw 0x200000 0x1E00000\\\\;" \
+"firmware.dfu raw 0x200000 0xE00000\\\\;" \
 "uboot-extra-env.dfu raw 0xFF000 0x1000\\\\;" \
 "uboot-env.dfu raw 0x100000 0x20000\\\\;" \
 "spare.dfu raw 0x120000 0xE0000\0" \
-"dfu_sf=gpio set 15;set stdout serial@e0001000;echo Entering DFU SF mode ... && run dfu_sf_info && dfu 0 sf 0:0:40000000:0 && if test -n ${dfu_alt_num} && test ${dfu_alt_num} = 1; "\
-"then set fit_size ${filesize} && set dfu_alt_num && env save; fi;gpio clear 15;\0"
+"dfu_sf=gpio set 42;set stdout serial@e0001000;echo Entering DFU SF mode ... && run dfu_sf_info && dfu 0 sf 0:0:40000000:0 && if test -n ${dfu_alt_num} && test ${dfu_alt_num} = 1; "\
+"then set fit_size ${filesize} && set dfu_alt_num && env save; fi;gpio clear 42;\0"
 
 
 /* NOR */
@@ -127,7 +127,7 @@
 	"dfu_ram_info=" \
 	"set dfu_alt_info " \
 	"dummy.dfu ram 0 0\\\\;" \
-	"firmware.dfu ram ${fit_load_address} 0x1E00000\0" \
+	"firmware.dfu ram ${fit_load_address} 0xE00000\0" \
 	"dfu_ram=echo Entering DFU RAM mode ... && run dfu_ram_info && dfu 0 ram 0\0" \
 	"thor_ram=run dfu_ram_info && thordown 0 ram 0\0"
 
@@ -327,7 +327,7 @@
 	"read_sf=sf probe 0:0 50000000 0 && run qspiboot_extraenv &&" \
 		"sf read ${fit_load_address} 0x200000 ${fit_size} && " \
 		"iminfo ${fit_load_address} || " \
-		"sf read ${fit_load_address} 0x200000  0x1E00000; \0" \
+		"sf read ${fit_load_address} 0x200000  0xE00000; \0" \
 	"ramboot_verbose=adi_hwref;echo Copying Linux from DFU to RAM... && " \
 		"run dfu_ram;" \
 		"if run adi_loadvals; then " \
@@ -342,7 +342,7 @@
 		"fi; " \
 		"envversion;setenv bootargs console=ttyPS0,115200 maxcpus=${maxcpus} rootfstype=ramfs root=/dev/ram0 rw earlyprintk clk_ignore_unused uboot=\"${uboot-version}\" && " \
 		"bootm ${fit_load_address}#${fit_config} || echo BOOT failed entering DFU mode ... && run dfu_sf \0" \
-	"qspiboot=set stdout nulldev;adi_hwref;test -n $PlutoRevA || gpio input 14 && set stdout serial@e0001000 && sf probe && sf protect lock 0 100000 && run dfu_sf;  " \
+	"qspiboot=set stdout nulldev;adi_hwref;test -n $PlutoRevA || gpio input 41 && set stdout serial@e0001000 && sf probe && sf protect lock 0 100000 && run dfu_sf;  " \
 		"set stdout serial@e0001000;" \
 		"itest *f8000258 == 480003 && run clear_reset_cause && run dfu_sf; " \
 		"itest *f8000258 == 480007 && run clear_reset_cause && run ramboot_verbose; " \
@@ -362,6 +362,22 @@
 			"echo Running uenvcmd ...; " \
 			"run uenvcmd; " \
 		"fi\0" \
+	"loaddfu=if mmc rescan; then " \
+		"fatload mmc 0 ${fit_load_address} antsdre200.dfu; " \
+		"sf probe; " \
+		"sf update ${fit_load_address} 0x200000 0xE00000; " \
+		"fatload mmc 0 0x20000 uboot-env.dfu; " \
+		"sf update 0x20000 0x100000 0x20000; " \
+		"bootm ${fit_load_address}#${fit_config};" \
+		"fi\0" \
+	"sdboot=if mmcinfo; then " \
+			"run uenvboot; " \
+			"echo Copying Linux from SD to RAM... && " \
+			"load mmc 0 ${fit_load_address} ${kernel_image} && " \
+			"load mmc 0 ${devicetree_load_address} ${devicetree_image} && " \
+			"load mmc 0 ${ramdisk_load_address} ${ramdisk_image} && " \
+			"bootm ${fit_load_address} ${ramdisk_load_address} ${devicetree_load_address}; " \
+		"fi\0" \
 	"usbboot=if usb start; then " \
 			"run uenvboot; " \
 			"echo Copying Linux from USB to RAM... && " \
